community/kernel/src/main/java/org/neo4j/kernel/recovery/PositionToRecoverFrom.java

```diff
@@ -47,18 +49,20 @@ public PositionToRecoverFrom( LatestCheckPointFinder checkPointFinder )
      * start recovery from
      * @throws IOException if log files cannot be read
      */
-    @Override
-    public LogPosition apply( long currentLogVersion ) throws IOException
+    public LogRecoveryInfo apply( long currentLogVersion ) throws IOException
```

Comment:
should we just return CheckPoint object here instead? it sounds reasonable that we have a checkpoint to be able to do recovery from them, so why not to just use them? wdyt?